### DISCUSSION TITLE
Can't run Nuclei with custom workflows in Github Action : no such file or directory.

### DISCUSSION THREAD

**Discussion by syhbt** [projectdiscovery/nuclei#4232] [Q&A] [answered]
Hi, I need help.

I tried running Nuclei on GitHub Action using nuclei custom workflow and nuclei custom template, everything works normally when I use nuclei custom template, but I get an error when I use my nuclei custom workflow. The error message indicated that my Nuclei custom workflow was not detected because Nuclei fetched the file in the default path.

I get error messages like this:
```
[ERR] Could not find template 'custom-workflows/nuclei-fuzzing-workflow.yaml': could not find file: stat /home/runner/nuclei-templates/custom-workflows/nuclei-fuzzing-workflow.yaml: no such file or directory
```
Complete error messages :
```
Run projectdiscovery/nuclei-action@main
Download and install Nuclei 2.9.15
/home/runner/work/_temp/b7d2048f-d797-666e-8609-636d7a290ef6/nuclei -target=http://testphp.vulnweb.com/listproducts.php?cat=1 -w=custom-workflows/nuclei-fuzzing-workflow.yaml -se=nuclei.sarif -o=nuclei.log -irr -rc=github-report.yaml

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v2.9.15

		projectdiscovery.io

[INF] nuclei-templates are not installed, installing...
[INF] Successfully installed nuclei-templates at /home/runner/nuclei-templates
[ERR] Could not find template 'custom-workflows/nuclei-fuzzing-workflow.yaml': could not find file: stat /home/runner/nuclei-templates/custom-workflows/nuclei-fuzzing-workflow.yaml: no such file or directory
[INF] Current nuclei version: v2.9.15 (latest)
[INF] Current nuclei-templates version: v9.6.4 (latest)
[INF] Targets loaded for current scan: 1
[INF] No results found. Better luck next time!
[FTL] Could not run nuclei: no valid templates were found
/home/runner/work/_actions/projectdiscovery/nuclei-action/main/dist/index.js:2348
                error = new Error(`The process '${this.toolPath}' failed with exit code ${this.processExitCode}`);
                        ^

Error: The process '/home/runner/work/_temp/b7d2048f-d797-666e-8609-636d7a290ef6/nuclei' failed with exit code 1
    at ExecState._setResult (/home/runner/work/_actions/projectdiscovery/nuclei-action/main/dist/index.js:2348:25)
    at ExecState.CheckComplete (/home/runner/work/_actions/projectdiscovery/nuclei-action/main/dist/index.js:2331:18)
    at ChildProcess.<anonymous> (/home/runner/work/_actions/projectdiscovery/nuclei-action/main/dist/index.js:2225:27)
    at ChildProcess.emit (node:events:513:28)
    at maybeClose (node:internal/child_process:1100:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:304:5)
```
this is my GitHub Action Workflow file (nuclei-test.yml):
```
      - name: Nuclei - Vulnerability Scan
        uses: projectdiscovery/nuclei-action@main
        with:
          target: http://testphp.vulnweb.com/listproducts.php?cat=1
          workflows: custom-workflows/nuclei-fuzzing-workflow.yaml
```
this is my Nuclei Workflow file (nuclei-fuzzing-workflow.yaml) :
```
id: nuclei-fuzzing-workflow
info:
  name: Nuclei Fuzzing Workflow
  author: me

workflows:
  - template: custom-templates/fuzzing/sql-fuzz.yaml
  - template: custom-templates/fuzzing/xss-fuzz.yaml
```

I have tested my workflow file (nuclei-fuzzing-workflow.yaml) on my local machine and it works fine, but I get an error when I run it on GitHub Action.

Is there something wrong with the steps I took? and how do I solve this problem?
Thank You.

**Comment by MetzinAround** [comment]
I think it's likely just a directory path issue since the original error is "could not find template". I'd double check how actions is trying to access that particular template and make sure that the directory path is correct. 

Let me know if you find anything, and I'll keep an eye on this to see what's up.

**Reply by syhbt** [reply, marked as answer]
Thank you for your response.  
I found the directory path issue again in the `-config` flag.  when the github action runs `-config custompath/myconfig.yaml` it results the same error message.  I tried running on my local machine and it worked without errors.  Then I realized that the nuclei on my local machine are running on the global path, while the nuclei on the github action are running on "non-global" path `/home/runner/work/_temp/a457-as455-45gkl-etc-etc/nuclei`. 
So I tried again on my local machine with nuclei located on a "non-global" path `/home/user/nucnuc/nuclei` and it resulted the same error as nuclei running on github action. 

In my conclusion, `-w` flag and `-config` flag can only work properly when installed in the global path. 

I assume this is a bug in the nuclei when running `-w` flag and `-config` flag considering that the `-t` flag and other flags that use custom paths work well without errors. 

For my problem with github action, my temporary solution was that I just moved nuclei from 
```/home/runner/work/_temp/a457-as455-45gkl-etc-etc/nuclei``` 
to
```${{ github.workspace }}/nuclei``` 
and it works fine without errors.

**Reply by MetzinAround** [reply]
Thank you for posting these details and for letting us know what your workaround is. I'll make sure our team sees this make sure they're aware of this possible bug.

**Comment by MetzinAround** [comment]
@syhbt  there's a PR that makes some changes now: https://github.com/projectdiscovery/nuclei-action/pull/68. Keep an eye out for when it's live. 

Thanks @dogancanbakir for the quick work!
